### DISCUSSION TITLE
Next boards to support

### DISCUSSION THREAD

**Discussion by ffenix113** [ffenix113/zigbee_home#18] [General]
Hello everyone!

While adding sensors is more straight forward I want to also add support for boards based on nRF52840 for people to use.

Currently I have two boards: nRF52840 Dongle and Arduino Nano 33 BLE. This allows to just have minimal diversity in board/software configurations for development.

I would like to know which boards(preferably with links to the board) people would be interested in getting support for, and, if possible, bootloader that those boards are using. It seems that bootloader present on the board is more important than the board itself for support, and if one board with some bootloader is supported - it is practically guaranteed that all other boards with same bootloader would be supported as well.

For reference on some boards: 
* @joric's list of custom keyboard board alternatives: https://github.com/joric/nrfmicro/wiki/Alternatives

**Comment by lorenzspenger** [comment]
Seeed Xiao BLE https://wiki.seeedstudio.com/XIAO_BLE would be neat :)

**Reply by koenvervloesem** [reply]
I agree that this is a good candidate. This shouldn't be that difficult, as it's already [supported by Zephyr](https://docs.zephyrproject.org/latest/boards/arm/xiao_ble/doc/index.html). It ships with the [Adafruit nRF52 Bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader).

**Reply by ffenix113** [reply]
Thanks for suggestion.

I am ironing out some issues, but I had nice progress with flashing ItsyBitsy nrf52840 bootloader on my Arduino Nano 33 BLE and successfully got firmware that was able to start and interface with my Zigbee2Mqtt instance.

Do you have that particular board, in case I may provide some firmware, or configuration file from which to generate firmware, to test on it?

**Reply by lorenzspenger** [reply]
I do have some of these boards at hand, yes

**Reply by ffenix113** [reply]
Nice. Thank you, but I also got myself such board, as I couldn't get my Arduino Nano 33 BLE with Adafruit bootloader to work on generated firmware.

So currently I am working on resolving the issue with the firmware configuration to work on Adafruit bootloader on Xiao and Nano 33.

**Comment by Hedda** [comment]
Please also look into adding support for E73-2G4M08S1C and E73-2G4M08S1CX nRF52840 radio modules from Ebyte (a.k.a. cdEbyte)

https://www.cdebyte.com/products/E73-2G4M08S1C

https://www.ebyte.com/en/downpdf.aspx?id=445

https://www.ebyte.com/product-view-news.html?id=444

E73-2G4M08S1CX  is an identical alternative to E73-2G4M08S1C, except for that E73-2G4M08S1CX requires an external antenna:

https://www.cdebyte.com/products/E73-2G4M08S1CX/4

https://www.ebyte.com/product-view-news.html?id=1317

If possible maybe also consider looking into the less expensive E73-2G4M08S1E and E73-2G4M08S1EX but those are based on nRF52833:

https://www.cdebyte.com/products/E73-2G4M08S1E

https://www.cdebyte.com/products/E73-2G4M08S1EX

https://www.ebyte.com/en/product-view-news.html?id=1035

https://www.ebyte.com/product-view-news.html?id=1034

These radio modules are used in the "b-parasite" open-source soil moisture sensor project for DIY Zigbee plant/flower sensors, see -> https://github.com/ffenix113/zigbee_home/discussions/5

I have no idea what bootloader these ship with but maybe the "b-parasite" project has more information on bootloader?

https://github.com/rbaron/b-parasite

It is very possible that radio modules from Ebyte ship with a custom bootloader and need an SWD programmer to flash it the first time.

E.g. first flashing the Adafruit nRF52 Bootloader with a JLink debug probe -> https://github.com/adafruit/Adafruit_nRF52_Bootloader 

PS: Might also consider looing into E104-BT5040U (and E104-BT5040UA) are USB dongles from eByte/CDeByte based on nRF52840:

https://www.ebyte.com/product-view-news.html?id=1030

https://www.ebyte.com/product-view-news.html?id=1206

**Comment by Hedda** [comment]
> Please also look into adding support for E73-2G4M08S1C and E73-2G4M08S1CX nRF52840 radio modules from Ebyte (a.k.a. cdEbyte)

FYI, popular alternatives to Ebyte radio modules are those from RF-star like these RF-BM-ND05, RF-BM-ND06, and RF-BM-ND05I:

https://www.szrfstar.com/product/tag/nrf52840-en

https://www.szrfstar.com/product/nrf52840_bluetooth_mesh_module_for_smart_home-en.html

https://www.szrfstar.com/product/nordic_nrf52840_mesh_module_RFBMND06-en.html

https://www.szrfstar.com/product/nordic_nrf52840_long_range_ble_module_with_ipex-en.html

RFstar also have a nRF52840 dongle:

https://www.szrfstar.com/product/nrf52840_bluetooth_zigbee_usb_dongle-en.html

**Comment by Hedda** [comment]
When targeting DIY hardware tinkerers it is probably a good idea to look at what nRF52840 based products that Adafruit is offering today:

https://www.adafruit.com/search?q=nRF52840+

Example:

Adafruit Feather nRF52840 Express

https://www.adafruit.com/product/4062

https://learn.adafruit.com/introducing-the-adafruit-nrf52840-feather/overview

Works out of the box with all Adafruit FeatherWings -> https://www.adafruit.com/categories/814

Adafruit Feather nRF52840 Sense

https://www.adafruit.com/product/4516

Adafruit LED Glasses Driver - nRF52840 Sensor Board

https://www.adafruit.com/product/5217

Adafruit ItsyBitsy nRF52840 Express - Bluetooth LE

https://www.adafruit.com/product/4481

Adafruit CLUE - nRF52840 Express with Bluetooth LE

https://www.adafruit.com/product/4500

nRF52840 USB Key with TinyUF2 Bootloader (Bluetooth Low Energy - MDBT50Q-RX)

https://www.adafruit.com/product/5199

PS: Adafruit is also selling their nRF52840 SMD (Surface Mount Device) RF-module that are used in all those board but those are not DIY-friendly to buy as stand-alone as need to use reflow soldering but might still be nice if they are supported:

https://www.adafruit.com/product/4078

**Comment by mitkoge** [comment]
Hi would you consider the range of "nice nano compatible" boards? Do they make sense?

**Reply by ffenix113** [reply]
For sure. Any suggested boards that you have in mind? Preferably with links to store pages.

**Reply by ffenix113** [reply]
I have ordered some boards, but it looks like they use Adafruit UF2 bootloader. This means that they are supported by this project, though I would need to update Wiki pages to indicate that and provide proper usage.

Thank you for suggestion!

**Comment by mitkoge** [comment]
sorry for bothering, i intendet to appy as a tester and those were the boards i have at hand.
[nice nano](https://nicekeyboards.com/nice-nano/)
[SuperMini](https://wiki.icbbuy.com/doku.php?id=developmentboard:nrf52840)

**Reply by ffenix113** [reply]
Oh, tester would be amazing indeed and highly appreciated! 

As I have mentioned [above](https://github.com/ffenix113/zigbee_home/discussions/18#discussioncomment-14740181) I have purchased some of the boards as well to test and use them in my home with firmwares created by this project.

I actually have 3 boards that are currently being used as different sensors, and a couple more for ongoing improvements.

About testing: I am working on usability improvements. This would include documentation, project executables, firmware build in docker and flashing to boards. Do you have in mind something that you would like to test?
